ui/app/helpers/utils/transactions.util.js

```diff
@@ -43,7 +58,19 @@ const registry = new MethodRegistry({ provider: global.ethereumProvider })
     const fourBytePrefix = prefixedData.slice(0, 10)
 
     try {
-      const sig = await registry.lookup(fourBytePrefix)
+      let fourByteSig
+      try {
+        fourByteSig = getMethodFrom4Byte(fourBytePrefix)
```

Comment:
I don't quite think that the `try` block will catch any async errors here, we might be able to chain a `catch` onto this here, e.g.,

```js
fourByteSig = getMethodFrom4Byte(fourBytePrefix).catch((e) => {
    log.error(e);
    return null;
})
```

Reply:
@whymarrh Could we rely on the existing try-catch block. So the whole thing would become:

```
    try {
      const fourByteSig = getMethodFrom4Byte(fourBytePrefix)

      let sig = await registry.lookup(fourBytePrefix)

      if (!sig) {
        sig = await fourByteSig
      }

      if (!sig) {
        return {}
      }

      const parsedResult = registry.parse(sig)

      return {
        name: parsedResult.name,
        params: parsedResult.args,
      }
    } catch (error) {
      log.error(error)
      const tokenData = getTokenData(data)
      const { name } = tokenData || {}
      return { name }
    }
```

Reply:
I think we still need to handle the case where the promise rejects, a case that isn't handled by the existing `try` block. In the event that `getMethodFrom4Byte(fourBytePrefix)` rejects and `registry.lookup(fourBytePrefix)` resolves to something truthy we don't have a handler for the rejected promise which I believe will result in [an unhandled promise rejection](https://stackoverflow.com/a/40500776/1267663).